scripts/utilities/retry.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-console */
+
+/**
+ * Retry - Retry a function on failure
+ *
+ * @param {function} Run          - The function to retry on failure
+ * @param {string}   errorMessage - Additional information when the limit is reached
+ * @param {number}   attempts     - The total number of attempts
+ * @param {number}   limit        - The maximum number of attempts
+ * @param {number}   delay        - The seconds to wait between attempts
+ */
+const Retry = function(Run, errorMessage, attempts = 0, limit = 3, delay = 10) {
+  // Check that there is a function to try
+  if (typeof Run !== 'function') {
```

Comment:
We don't do defensive programming, see https://github.com/Shopify/ruby-style-guide#general

That's the ruby styleguide but it applies generally to programming at Shopify

Reply:
Sounds good to me, this is just me used to writing tests. Should I be writing tests for this?

Reply:
Hmm, I want to say that we should write at least some simple tests but I don't know if we even have testing set up for scripts right now. Let me check and get back to you, but don't worry about it for now

Reply:
Thanks mate :)